ncm-iptables/src/main/perl/iptables.pm

```diff
@@ -636,101 +611,84 @@ sub rule_options_translate {
-	    next if ( ! defined $iptables->{$table} || $iptables->{$table} eq "" ||
-		      ref($iptables->{$table}) !~ /^HASH/ );
-
-	    print FILE "*$table\n";
-
-
-	    if ( defined $iptables->{$table}->{preamble} &&
-		 ref($iptables->{$table}->{preamble}) =~ /^HASH/ ) {
-		my $preamble = $iptables->{$table}->{preamble};
-
-
-
-		foreach $chain ( @{$iptables_totality{$table}{chains}}) {
-		    next if ( ! defined $preamble->{$chain} || $preamble->{$chain} eq "" );
-		    my $g = $chain;
-		    $g =~ tr/a-z/A-Z/;
-		    $preamble->{$chain} =~ s/^[\s\t]*|[\s\t]*$//g;
-		    $preamble->{$chain} =~ s/[\s\t+]/ /g;
-		    print FILE ":$g $preamble->{$chain}\n";
-		}
-	    }
-
-
-
-	    foreach $target ( sort keys %{$iptables_totality{$table}{user_targets}} ){
-		print FILE "-N $target\n";
-	    }
-
-	    foreach $target (@{$iptables_totality{$table}{targets}}) {
-
-		next if ( ! defined $iptables->{$table}->{rules}->{$target}         );
-		next if (   ref($iptables->{$table}->{rules}->{$target}) !~ /^HASH/ );
-		next if ( ! scalar(%{$iptables->{$table}->{rules}->{$target}})      );
-
-		foreach $name (sort { $a <=> $b; } keys %{$iptables->{$table}->{rules}->{$target}}) {
-
-		    next if ( $name !~ /^\d+$/ );
-
-		    $rule = $iptables->{$table}->{rules}->{$target}->{$name};
-		    $line = '';
-		    foreach $field (&sort_keys($self,$rule)) {
-			$line .= ($line) ? " $field" : $field;
-			$line .= " $rule->{$field}" if $options_arg{$field};
-		    }
-		    print FILE "$line\n" if $line and $line !~ /^-N/;
-
-		}
-
-	    }
-
-	    print FILE "$iptables->{$table}->{epilogue}\n"
-		if ( defined $iptables->{$table}->{epilogue} && $iptables->{$table}->{epilogue} ne "");
-
-	}
-    }
-
-    # Close the temporary file.
-    unless ( close(FILE) ) {
-	$? = 7;
-	$@ = "cannot close $filename";
-	return $?;
-    }
-
-    $? = 0;
-    $@ = "modified $filename";
-    return $?;
+    if (defined $iptables && ref($iptables) =~ /^HASH/) {
+        $self->debug(5, "iterating over tables");
+        foreach my $table (keys %iptables_totality) {
+            $self->debug(5, "processing table $table");
+            next if (!$iptables->{$table} || ref($iptables->{$table}) !~ /^HASH/);
+            print $fh "*$table\n";
+
+            if (defined $iptables->{$table}->{preamble} && ref($iptables->{$table}->{preamble}) =~ /^HASH/ ) {
+                my $preamble = $iptables->{$table}->{preamble};
+                $self->debug(5, "table has preamble $preamble");
+
+                foreach my $chain (@{$iptables_totality{$table}{chains}}) {
+                    $self->debug(5, "processing chain $chain");
+                    next if (!$preamble->{$chain});
+                    my $g = $chain;
+                    $g =~ tr/a-z/A-Z/;
+                    $preamble->{$chain} = $self->trim_whitespace($preamble->{$chain});
+                    $preamble->{$chain} = $self->collapse_whitespace($preamble->{$chain});
+                    print $fh ":$g $preamble->{$chain}\n";
```

Comment:
remove the `$g`, do a `print $fh ":",uc($chain), " ", $preamble->{$chain}, "\n";`?